cmd/pkg/baseline/generator_oscal.go

```diff
@@ -0,0 +1,120 @@
+// SPDX-FileCopyrightText: Copyright 2025 The OSPS Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package baseline
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	oscal "github.com/defenseunicorns/go-oscal/src/types/oscal-1-1-3"
+	"github.com/ossf/security-baseline/pkg/types"
+)
+
+const (
+	VersionOSPS = "devel"
+	controlHREF = "https://baseline.openssf.org/versions/%s#%s"
+	catalogUUID = "8c222a23-fc7e-4ad8-b6dd-289014f07a9f"
+
+	// OpenSSFNS is the OSCAL namespace URI to define the baseline names.
+	OpenSSFNS = "http://baseline.openssf.org/ns/oscal"
+)
+
+func (g *Generator) ExportOSCAL(b *types.Baseline, w io.Writer) error {
+	n := time.Now()
+	catalog := oscal.Catalog{
+		UUID:   catalogUUID,
+		Groups: nil,
+		Metadata: oscal.Metadata{
+			LastModified: n,
+			Links: &[]oscal.Link{
+				{
+					Href: fmt.Sprintf(controlHREF, VersionOSPS, ""),
+					Rel:  "canonical",
+				},
+			},
+			OscalVersion: "1.1.3",
+			Published:    &n,
+			Title:        "Open Source Project Security Baseline",
+			Version:      VersionOSPS,
+		},
+	}
+
+	catalogGroups := []oscal.Group{}
+
+	for code, cat := range b.Categories {
+		group := oscal.Group{
+			Class:    "OSPS",
+			Controls: nil,
+			ID:       code,
+			Title:    cat.Description,
+		}
+
+		controls := []oscal.Control{}
+
+		for _, control := range cat.Controls {
+
+			parts := []oscal.Part{}
+			for _, ar := range control.Requirements {
+				parts = append(parts, oscal.Part{
+					Class: control.ID,
+					ID:    ar.ID,
+					Name:  ar.ID,
+					Ns:    "",
+					Parts: &[]oscal.Part{
+						{
+							ID:    ar.ID + ".R",
+							Name:  "recomemendation",
+							Ns:    OpenSSFNS,
```

Comment:
I am going to try and dig into this over the weekend, but in the short-term you may want to correct the spelling of the part name.

```suggestion
							Name:  "recommendation",
							Ns:    OpenSSFNS,
```

I would say a higher-level issue from using [the maintained `oscal-cli` community fork](https://github.com/metaschema-framework/oscal-cli/) is that there may be specific OSCAL rules that say a part should have a certain name if it is in the "default" core NIST OSCAL namespace. Weirdly, yours is not. You did it right. However, the error still persists in the tooling, so I will look into that; it might be a bug on that end so you can safely ignore it. :-)

Reply:
Actually it seems it is related to the NIST's definition in the OSCAL model, not the tooling. More to follow after https://github.com/usnistgov/OSCAL/issues/2112 is resolved and released in a new model version. You helped find an interesting edge case!

Reply:
I noticed it too, it also seems that it is ignoring namespaced names.